### DISCUSSION TITLE
[v4] Trouble updating to latest (from alpha 17) tailwindcss + @tailwindcss/postcss in a Laravel (vite) app

### DISCUSSION THREAD

**Discussion by nathangross** [tailwindlabs/tailwindcss#14652] [Help]
I have been using Tailwind v4 (postcss) through alpha 17 just fine. It seems like something changed that is breaking my setup—but I can't figure out what it was. The release notes for the alphas were pretty sparse in the earlier alphas.

When I try to update these to any alpha after 17

```
"@tailwindcss/postcss": "^4.0.0-alpha.26",
"tailwindcss": "^4.0.0-alpha.26"
```

I get this error:

```
[vite] Internal server error: [postcss] (0 , p.compile) is not a function
  Plugin: vite:css
```

Any ideas?

**Comment by philipp-spiess** [comment]
@nathangross Hey! This issue usually indicates that one of the dependencies is not on the right version (we had a breaking change in one of the internal APIs in previous alpha versions, so it's important all of the alpha releases match up). Can you check what your resolved versions of `tailwindcss` and `@tailwindcss/postcss` point to by doing something like this?

```bash
cat node_modules/tailwindcss/package.json 
cat node_modules/@tailwindcss/postcss/package.json 
```

Alternatively you could take a look at the lockfile of your package manager.

**Comment by nathangross** [comment]
@philipp-spiess Thanks for the reply—and sorry for my late response!

My project is perhaps a bit non-standard. I'm using **v4 on the frontend** and **v3.4 on the backend**. I've been handling this with Yarn workspaces (also trying out pnpm) fine through I think v4 around Alpha 17 or so. 

```
my-project/packages/frontend/[node_modules, package.json, postcss.config.js] (v4)
my-project/packages/backend/[node_modules, package.json, postcss.config.js] (v3.4)
```

Once I updated anything beyond that 17 something seemed to change that broke this specific setup (I think I saw a postcss-import package go away in the package lock diff?)

I've since had better luck with tailwind 4 with a more standard setup for the v4 stuff

```
my-project/[node_modules, package.json, postcss.config.js] (v4)
my-project/packages/backend/[node_modules, package.json, postcss.config.js] (v3.4)
```

That said, I wonder if there was something I could configure to get the workspaces to work nice with v4?

Hopefully that all makes sense. Holler if I can clarify!
